core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -90,6 +90,12 @@ private TableProperties() {
   public static final String ORC_VECTORIZATION_ENABLED = "read.orc.vectorization.enabled";
   public static final boolean ORC_VECTORIZATION_ENABLED_DEFAULT = false;
 
+  public static final String LOCALITY_ENABLED = "read.locality.enabled";
+  public static final String LOCALITY_ENABLED_DEFAULT = null;
```

Comment:
What is the current default? Is that inconsistent across uses and that's why this is null?

Reply:
> Is that inconsistent across uses and that's why this is null?

Yes. The current default comes from `LOCALITY_WHITELIST_FS.contains(scheme)`, which is true on `hdfs` and false otherwise.